### PR TITLE
Extend Python API with ScatterNDUpdate-3

### PR DESCRIPTION
**Pull request by kk271kg**
### Details:
 - src/bindings/python/src/openvino/runtime/opset4/ops.py: Implement ScatterNDUpdate operation
 - src/bindings/python/src/openvino/runtime/opset4/__init__.py: Import ScatterNDUpdate module
 - src/bindings/python/src/openvino/runtime/opset5/__init__.py: Import ScatterNDUpdate module
 - src/bindings/python/src/openvino/runtime/opset6/__init__.py: Import ScatterNDUpdate module
 - src/bindings/python/src/openvino/runtime/opset7/__init__.py: Import ScatterNDUpdate module
 - src/bindings/python/src/openvino/runtime/opset8/__init__.py: Import ScatterNDUpdate module
 - src/bindings/python/src/openvino/runtime/opset9/__init__.py: Import ScatterNDUpdate module
 - src/bindings/python/src/openvino/runtime/opset10/__init__.py: Import ScatterNDUpdate module
 - src/bindings/python/src/openvino/runtime/opset11/__init__.py: Import ScatterNDUpdate module
 - src/bindings/python/src/openvino/runtime/opset12/__init__.py: Import ScatterNDUpdate module
 - src/bindings/python/src/openvino/runtime/opset13/__init__.py: Import ScatterNDUpdate module
 - src/bindings/python/tests/test_graph/test_ops_scatter_nd_update.py: New file created to test the functionalities of ScatterNDUpdate implementation

### Tickets:
 - 121506 -  [Good First Issue]: Extend Python API with ScatterNDUpdate-3 #20284

### CODE REVIEW

**Comment by p-wysocki** [comment]
build_jenkins

**Review by jiwaszki — COMMENTED** [review]
From Python point of view everything is good. Amazing job at writing tests! 🥇 

However I have some concerns about exposing it in opset-13. It could be dated back to opset3 (or 4?) and updated accordingly across all of them. @mitruska can you share your opinions on that?

Is there some misleading info in C++ as well? `v3` and `opset4` afterwards:
https://github.com/openvinotoolkit/openvino/blob/28208c6c9a22c8ba236391239a7feb7bce82cf00/src/core/include/openvino/op/scatter_nd_update.hpp#L9-L16

File: src/bindings/python/tests/test_graph/test_ops_scatter_nd_update.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+# Copyright (C) 2018-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+import pytest
+
+from openvino.runtime import PartialShape
+
+import openvino.runtime.opset13 as ov
+from openvino.runtime import Type
```
Comment (akuporos) [inline src/bindings/python/tests/test_graph/test_ops_scatter_nd_update.py:11]: I suggest to delete this line and import `Type` together with `PartialShape`

**Review by mitruska — COMMENTED** [review]
Hello, thank you @kk271kg for your contribution!

As @jiwaszki asked, I've left few suggestions regarding the alignment of the opset number with C++ API. `ScatterNDUpdate` is a special case, with the version of the op not the same as the opset for backward compatibility reasons.

File: src/bindings/python/tests/test_graph/test_ops_scatter_nd_update.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2018-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+import pytest
+
+from openvino.runtime import PartialShape, Type
```
Comment (akuporos) [inline src/bindings/python/tests/test_graph/test_ops_scatter_nd_update.py:8]:
```suggestion
from openvino import PartialShape, Type
```

Comment (akuporos) [reply src/bindings/python/tests/test_graph/test_ops_scatter_nd_update.py:8]: sometime  ago we introduced new imports, so it'd be better to use new namespace :)

File: src/bindings/python/src/openvino/runtime/opset4/ops.py [hunk 3]
```diff
@@ -442,3 +442,23 @@ def range(
...
+
+
+@nameable_op
+def scatter_nd_update(
+    data: NodeInput,
+    indices: NodeInput,
+    updates: NodeInput,
+    name: Optional[str] = None,
+) -> Node:
+    """"Return a node which performs ScatterNDUpdate.
```
Comment (jiwaszki) [inline src/bindings/python/src/openvino/runtime/opset4/ops.py:454]:
```suggestion
    """Return a node which performs ScatterNDUpdate.
```

Comment (mitruska) [inline src/bindings/python/src/openvino/runtime/opset4/ops.py:454]:
```suggestion
    """Return a node which performs ScatterNDUpdate.
```

**Review by jiwaszki — APPROVED** [review]
Only one remark that is also reported by our checks. Overall LGTM.

**Review by mitruska — APPROVED** [review]

**Review by akuporos — APPROVED** [review]

**Comment by akuporos** [comment]
build_jenkins

**Comment by akuporos** [comment]
Hi @kk271kg,

Unfortunately `test_scatter_nd_update_empty_indices_and_updates():` failed on windows job.
The line where it failed is `np.testing.assert_array_equal(result, expected)`.

>            arguments = self._arguments_as_outputs(arguments)
>            node = self.factory.create(op_type_name, arguments, attributes)
>  E       RuntimeError: Check 'element::Type::merge(result_et, node->get_input_element_type(0), node-> 
>  get_input_element_type(1))' failed at src\core\src\op\util\elementwise_args.cpp:25:
> E       While validating node 'opset1::Equal Equal_17580 (ScatterNDUpdate_17578[0]:i64[3], Constant_17579[0]:i32[]) -> ()' with friendly_name 'Equal_17580':
> E       Arguments do not have the same element type (arg0 element type: i64, arg1 element type: i32).
> ..\..\..\..\..\..\install\install_pkg\python\openvino\runtime\utils\node_factory.py:59: RuntimeError

**Review by p-wysocki — APPROVED** [review]

**Comment by akuporos** [comment]
build_jenkins

**Pull request merged**